source/extensions/filters/http/ratelimit/ratelimit.h

```diff
@@ -37,6 +37,39 @@ enum class FilterRequestType { Internal, External, Both };
  */
 enum class VhRateLimitOptions { Override, Include, Ignore };
 
+/**
+ * Rate limit logging information stored in the filter state for access logging.
+ */
+class RateLimitLoggingInfo : public Envoy::StreamInfo::FilterState::Object {
+public:
+  RateLimitLoggingInfo() {}
+
+  absl::optional<std::chrono::microseconds> latency() const { return latency_; };
+  absl::optional<uint64_t> bytesSent() const { return bytes_sent_; }
+  absl::optional<uint64_t> bytesReceived() const { return bytes_received_; }
+  Upstream::ClusterInfoConstSharedPtr clusterInfo() const { return cluster_info_; }
+  Upstream::HostDescriptionConstSharedPtr upstreamHost() const { return upstream_host_; }
+
+  void setLatency(std::chrono::microseconds ms) { latency_ = ms; };
+  void setBytesSent(uint64_t bytes_sent) { bytes_sent_ = bytes_sent; }
+  void setBytesReceived(uint64_t bytes_received) { bytes_received_ = bytes_received; }
+  void setClusterInfo(Upstream::ClusterInfoConstSharedPtr cluster_info) {
+    cluster_info_ = std::move(cluster_info);
+  }
+  void setUpstreamHost(Upstream::HostDescriptionConstSharedPtr upstream_host) {
+    upstream_host_ = std::move(upstream_host);
+  }
+
+  bool hasFieldSupport() const override { return true; }
+
+private:
+  absl::optional<std::chrono::microseconds> latency_;
+  absl::optional<uint64_t> bytes_sent_;
+  absl::optional<uint64_t> bytes_received_;
+  Upstream::ClusterInfoConstSharedPtr cluster_info_;
+  Upstream::HostDescriptionConstSharedPtr upstream_host_;
+};
```

Comment:
I didn't see how this works for logging. And I think this actually expose too much info which have exceeded the original requirements of https://github.com/envoyproxy/envoy/issues/39018. In this complex system, I think don't do something is more important than do something.